creator/data_templates/models.py

```diff
@@ -174,8 +186,10 @@ def released(self):
 
     def clean(self):
         """
-        Validate this template version
-
-        TODO Validate field definitions
+        Validate template version
         """
-        pass
+        # Try to fix any human errors in the data before validating against
+        # the field definitions schema (See FieldDefinitionsSchema.clean)
+        self.field_definitions = (
+            self.field_definitions_schema.load(self.field_definitions)
```

Comment:
this happens on upload right? so we never save the file they actually give us just the new cleaned version?
Not for now just wondering for later - is it possible/easy to make a report of the values that were changed?

Reply:
> so we never save the file they actually give us just the new cleaned version

Correct

> is it possible/easy to make a report of the values that were changed

yea i was thinking about this too. we could determine the values that changed in this mutation and report it back to them in the response. but perhaps more important in the near term..we may want to determine if any fields changed at all in the create new template version mutation bc we likely don't want to create a new template version unless the fields changed in some way